tests/components/smlight/test_config_flow.py

```diff
@@ -363,3 +363,138 @@ async def test_zeroconf_legacy_mac(
 
     assert len(mock_setup_entry.mock_calls) == 1
     assert len(mock_smlight_client.get_info.mock_calls) == 2
+
+
+async def test_reauth_flow(
+    hass: HomeAssistant,
+    mock_smlight_client: MagicMock,
+    mock_config_entry: MockConfigEntry,
+    mock_setup_entry: AsyncMock,
+) -> None:
+    """Test reauth flow completes successfully."""
+    mock_smlight_client.check_auth_needed.return_value = True
+    mock_config_entry.add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={
+            "source": SOURCE_REAUTH,
+            "entry_id": mock_config_entry.entry_id,
+        },
+        data=mock_config_entry.data,
+    )
```

Comment:
we have a new helper for this. `mock_config_entry.somethingsomethingreauth` (forgot the name)